homeassistant/components/greeneye_monitor/sensor.py

```diff
@@ -224,66 +257,79 @@ def _seconds_per_time_unit(self):
         if self._time_unit == TIME_HOURS:
             return 3600
 
+        # Config schema should have ensured it is one of the above values
+        raise Exception(
+            f"Invalid value for time unit: {self._time_unit}. Expected one of {TIME_SECONDS}, {TIME_MINUTES}, or {TIME_HOURS}"
+        )
+
     @property
-    def native_unit_of_measurement(self):
+    def native_unit_of_measurement(self) -> str:
         """Return the unit of measurement for this pulse counter."""
         return f"{self._counted_quantity}/{self._time_unit}"
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, Any] | None:
         """Return total pulses in the data dictionary."""
         if not self._sensor:
             return None
 
         return {DATA_PULSES: self._sensor.pulses}
 
 
-class TemperatureSensor(GEMSensor):
+class TemperatureSensor(GEMSensor[greeneye.monitor.TemperatureSensor]):
     """Entity showing temperature from one temperature sensor."""
 
     _attr_device_class = DEVICE_CLASS_TEMPERATURE
     _attr_icon = TEMPERATURE_ICON
 
-    def __init__(self, monitor_serial_number, number, name, unit):
+    def __init__(
+        self, monitor_serial_number: int, number: int, name: str, unit: str
+    ) -> None:
         """Construct the entity."""
         super().__init__(monitor_serial_number, name, "temp", number)
         self._unit = unit
 
-    def _get_sensor(self, monitor):
+    def _get_sensor(
+        self, monitor: greeneye.monitor.Monitor
+    ) -> greeneye.monitor.TemperatureSensor:
         return monitor.temperature_sensors[self._number - 1]
 
     @property
-    def native_value(self):
+    def native_value(self) -> float | None:
         """Return the current temperature being reported by this sensor."""
         if not self._sensor:
             return None
 
+        assert isinstance(self._sensor.temperature, float)
```

Comment:
If we know the type is a certain type we can use `typing.cast` when returning it.